gst/droidcodec/gstdroidvdec.c

```diff
@@ -1029,28 +1035,34 @@ gst_droidvdec_handle_frame (GstVideoDecoder * decoder,
   droid_media_codec_queue (dec->codec, &data, &cb);
   GST_VIDEO_DECODER_STREAM_LOCK (decoder);
   GST_LOG_OBJECT (dec, "acquired stream lock");
-  /* from now on decoder owns a frame reference so we cannot use the out label otherwise
-   * we will drop the needed reference
-   */
+
+  /* from now on decoder owns a frame reference */
 
   if (dec->downstream_flow_ret != GST_FLOW_OK) {
     GST_DEBUG_OBJECT (dec, "not handling frame in error state: %s",
         gst_flow_get_name (dec->downstream_flow_ret));
     ret = dec->downstream_flow_ret;
-    goto out;
+    goto unref;
   }
 
   g_mutex_lock (&dec->state_lock);
   if (dec->state == GST_DROID_VDEC_STATE_EOS) {
     GST_WARNING_OBJECT (dec, "got frame in eos state");
     g_mutex_unlock (&dec->state_lock);
     ret = GST_FLOW_EOS;
-    goto out;
+    goto unref;
   }
+
   g_mutex_unlock (&dec->state_lock);
 
   ret = GST_FLOW_OK;
 
+  goto unref;
+
+unref:
```

Comment:
It's the first time ever I saw `goto` used like that (to jump across an empty line).